shared/actions/kbfs/index.js

```diff
@@ -1,33 +1,47 @@
 // @flow
 import * as Constants from '../../constants/kbfs'
-import type {AsyncAction} from '../../constants/types/flux'
-import {fsListRpc} from '../../constants/types/flow-types'
-import {openInKBFS as platformOpenInKBFS} from './index.platform'
+import {fsListRpcPromise} from '../../constants/types/flow-types'
+// $FlowIssue flow doens't understand generator exports
+import {openSaga} from './index.platform'
+import {call, put} from 'redux-saga/effects'
+import {takeLatest, takeEvery} from 'redux-saga'
 
-export function fsList (path: string) : AsyncAction {
-  return (dispatch) => {
-    fsListRpc({
-      param: {path},
-      callback: (error, result) => {
-        console.log('fs.List: ', error, result)
-        if (error) {
-          dispatch({
-            type: Constants.fsList,
-            payload: error,
-            error: true,
-          })
-        } else {
-          dispatch({
-            type: Constants.fsList,
-            payload: result,
-            error: false,
-          })
-        }
-      },
-    })
+import type {ListResult} from '../../constants/types/flow-types'
+import type {FSList, FSListed, FSOpen} from '../../constants/kbfs'
+import type {SagaGenerator} from '../../constants/types/saga'
+
+function fsList (path: string) : FSList {
+  return {type: Constants.fsList, payload: {path}}
+}
+
+function openInKBFS (path: string = ''): FSOpen {
+  return {type: Constants.fsOpen, payload: {path}}
+}
+
+function * _listSaga (action: FSList): SagaGenerator<any, any> {
```

Comment:
this isn't actually consumed by anything (as before). I think @gabriel added this during the kbfs mobile stuff so its not really hooked up / used anywhere yet